packages/flutter/lib/src/widgets/will_pop_scope.dart

```diff
@@ -9,10 +9,87 @@ import 'routes.dart';
 /// Registers a callback to veto attempts by the user to dismiss the enclosing
 /// [ModalRoute].
 ///
+/// {@tool snippet --template=stateful_widget}
+///
+/// Whenever the back button is pressed, you will get a callback at [onWillPop],
+/// which returns a [Future]. If the [Future] returns true, the screen is
+/// popped.
+///
+/// ```dart
+/// bool shouldPop = true;
+/// @override
+/// Widget build(BuildContext) {
+///   return WillPopScope (
+///     onWillPop: () async {
+///       return shouldPop;
+///     },
+///     child: Center(
+///     ),
```

Comment:
Consider changing this to a `Text('WillPopScope sample')` so that something is displayed on screen if this is copy-pasted.